### PR TITLE
Token check

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/7022

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
covered with tests

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_notification_service_impl.cc [hunk 1]
```diff
@@ -377,7 +377,8 @@ void RewardsNotificationServiceImpl::OnReconcileComplete(
   if ((converted_result == ledger::Result::LEDGER_OK &&
        converted_type == ledger::RewardsType::AUTO_CONTRIBUTE) ||
        converted_result == ledger::Result::LEDGER_ERROR ||
-       converted_result == ledger::Result::NOT_ENOUGH_FUNDS ||
+      (converted_result == ledger::Result::NOT_ENOUGH_FUNDS &&
+       converted_type != ledger::RewardsType::ONE_TIME_TIP)||
```
Comment (masparrow) [inline components/brave_rewards/browser/rewards_notification_service_impl.cc]: nit: Might be an opportunity to break this up into something more readable?
```
bool canAutoContribute = (converted_result == ledger::Result::LEDGER_OK) &
       (converted_type == ledger::RewardsType::AUTO_CONTRIBUTE);
bool insufficientFundsForContribution = (converted_result == ledger::Result::NOT_ENOUGH_FUNDS) &
       (converted_type != ledger::RewardsType::ONE_TIME_TIP);

if (canAutoContribute || 
converted_result == ledger::Result::LEDGER_ERROR || 
insufficientFundsForContribution ||
converted_result == ledger::Result::TIP_ERROR) {
```
Or even bring each condition out into its own bool... just a thought.

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_notification_service_impl.cc]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_client_mock.cc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/ledger_client_mock.h"
+
+namespace ledger {
+
+MockLogStreamImpl::MockLogStreamImpl(
+    const char* file,
+    const int line,
+    const ledger::LogLevel log_level) {
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/ledger_client_mock.cc]: `ledger::` not required

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_client_mock.h [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+#include "testing/gmock/include/gmock/gmock.h"
+
+namespace ledger {
+
+class MockLogStreamImpl : public ledger::LogStream {
+ public:
+  MockLogStreamImpl(
+      const char* file,
+      int line,
+      const ledger::LogLevel log_level);
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/ledger_client_mock.h]: ledger namespace not required

File: vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc [hunk 4]
```diff
@@ -137,9 +137,16 @@ void Unblinded::OnUnblindedTokens(
   }
 
   const auto reconcile = ledger_->GetReconcileById(viewing_id);
+  const uint64_t now = static_cast<uint64_t>(base::Time::Now().ToDoubleT());
   double current_amount = 0.0;
   ledger::UnblindedTokenList token_list;
+  std::vector<std::string> delete_list;
   for (auto & item : list) {
+    if (item->expires_at < now) {
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc]: we need to check this only if expires_at is > then 0. Fix it

Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc]: we need to check this only if expires_at is > then 0. Fix it

[the remark this answers is not in the document]
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc]: done

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc [hunk 5]
```diff
@@ -87,6 +87,16 @@ void GenerateSuggestion(
...
 }
 
+bool HasTokenExpired(ledger::UnblindedTokenPtr token) {
+  if (!token) {
+    return true;
+  }
+
+  const uint64_t now = static_cast<uint64_t>(base::Time::Now().ToDoubleT());
+
+  return token->expires_at > 0 && token->expires_at < now;
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc]: nit: now should be checked with `<=`

**Pull request merged**